felix/rules/nat.go

```diff
@@ -58,6 +58,11 @@ func (r *DefaultRuleRenderer) makeNATOutgoingRuleIPTables(ipVersion uint8, proto
 		SourceIPSet(masqIPsSetName).
 		NotDestIPSet(allIPsSetName)
 
+	if r.Config.HostSubnetNATExclusion {
```

Comment:
You are using the all-host IP set so the config param name is misleading since it's not using the subnet at all.